docs/BUILD.md

```diff
@@ -0,0 +1,128 @@
+required to run soci-snapshotter; to confirm please check with `sudo ctr version`.
+- **fuse** - used for mounting without root access (`sudo yum install fuse`).
+- **zlib** - used for decompression and ztoc creation; soci builds zlib statically into its binaries
+(`sudo yum install zlib-devel zlib-static`).
+- **gcc** - used for compiling SOCI's c code, gzip's zinfo implementation (`sudo yum install gcc`).
+- **[flatc](https://github.com/google/flatbuffers)** - used for compiling ztoc
+flatbuffer file and generating corresponding go code.
+
+For fuse/zlib/gcc, they can be installed by your Linux package manager (e.g., `yum` or `apt-get`).
+
+For flatc, you can download and install a [release](https://github.com/google/flatbuffers/releases)
+into your `/usr/local` (or other `$PATH`) directory. For example:
+
+```shell
+wget -c https://github.com/google/flatbuffers/releases/download/v23.3.3/Linux.flatc.binary.g++-10.zip
+sudo unzip Linux.flatc.binary.g++-10.zip -d /usr/local
+rm Linux.flatc.binary.g++-10.zip
+```
+
+## Build soci-snapshotter
+
+First you need `git` to clone the repository (if you intend to contribute, you
+can fork the repository and clone your own fork):
+
+```shell
+git clone https://github.com/awslabs/soci-snapshotter.git
+cd soci-snapshotter
+```
+
+soci-snapshotter uses `make` as the build tool. Assuming you're in the root directory
+of the repository, you can build soci-snapshotter by:
+
+```shell
+make
+```
+
+This builds the project binaries into the `./out` directory. You can install them
+to a `PATH` directory (`/usr/local/bin`) with:
+
+```shell
+sudo make install
+# check soci can be found in PATH
+sudo soci --help
+```
+
+When changing the ztoc's flatbuffer definition, you need to regenerate the generated
+code package with:
+
+> It's rare to make such a change, especially delete a field which is a breaking
+> change and discouraged by flatbuffers.
+
+```shell
+make flatc
+```
+
+## Test soci-snapshotter
+
+We have unit tests and integration tests as part of our automated CI, as well as
+benchmark tests that can be used to test the performance of soci-snapshotter. You
+can run these tests using the following `Makefile` targets:
+
+- `make test`: run all unit tests.
+- `make integration`: run all integration tests.
+- `make benchmarks` (experimental): run all benchmark tests. This requires some
+setup and preparation for public images with SOCI index which are used for benchmarking.
+It is tracked in [#245](https://github.com/awslabs/soci-snapshotter/issues/245).
+
+To speed up develop-test cycle, you can run individual test(s) by utilizing `go test`'s
+`-run` flag. For example, suppose you only want to run a test named `TestFooBar`, you can:
+
+```shell
+# 1. if TestFooBar is a unit test
+go test -run "TestFooBar" ./...
+
+# 2. if TestFooBar is an integration test
+ENABLE_INTEGRATION_TEST=true SOCI_SNAPSHOTTER_PROJECT_ROOT=$(pwd) go test -v -timeout=0 -run "TestFooBar" ./integration
+```
```

Comment:
`GO_TEST_FLAGS="-run LazyPullWithSparseIndex" make integration` will take advantage of other environment setup done by the Makefile, including recompiling the binaries if appropriate.

Also `-count=1` as a go test flag will disable caching. https://stackoverflow.com/questions/73432741/why-does-count-1-ignores-caching-in-go-tests

Reply:
Changed to `GO_TEST_FLAGS`